src/protocol.rs

```diff
@@ -195,7 +195,8 @@ impl SignatureShare {
             }
             (Some(prev_r), Some(new_r)) => {
                 if prev_r != new_r {
-                    return Err(InternalError::SignatureInstantiationError);
+                    error!("Failed to create signature as prev_r != new_r");
+                    return Err(InternalError::InternalInvariantFailed);
```

Comment:
suggestion: can we phrase this a bit more humanely? I think this is true:
> Failed to chain signature shares together because r-values were different. Got {prev_r}, expected {new_r}.

Also, I think this is a `ProtocolError`, since it's caused by a bad signature share from one of the parties.